docs/tour/add-features.mdx

```diff
@@ -26,70 +26,36 @@ This tutorial assumes you're following directly from the previous tutorial. If y
 <Tabs groupId="lang" queryString>
   <TabItem value="tinygo" label="Go">
 
-Let's extend this application to do more than just say "Hello!" 
+Let's extend this application to do more than just say "Hello!"
 
-Using the `PathWithQuery` method on the incoming request, we can check the request for a name provided in a query string, and then return a greeting with that name. If there isn't one or the path isn't in the format we expect, we'll default to saying "Hello, World!" 
-
-To make this code more readable, we'll add a helper function to extract the name from the path.
+Using the [Query](https://pkg.go.dev/net/url#URL.Query) method on the incoming request URL, we can check the request for a name provided in a query string, and then return a greeting with that name. If there isn't one or the path isn't in the format we expect, we'll default to saying "Hello, World!"
 
 ```go
-package main
-
 import (
-  "fmt"
-  "strings"
-  http "github.com/wasmcloud/wasmcloud/examples/golang/components/http-hello-world/gen"
-)
-
-// Helper type aliases to make code more readable
-type HttpRequest = http.ExportsWasiHttp0_2_0_IncomingHandlerIncomingRequest
-type HttpResponseWriter = http.ExportsWasiHttp0_2_0_IncomingHandlerResponseOutparam
-type HttpOutgoingResponse = http.WasiHttp0_2_0_TypesOutgoingResponse
-type HttpError = http.WasiHttp0_2_0_TypesErrorCode
+	"fmt" // [!code ++]
+	"net/http"
 
-type HttpServer struct{}
+	"go.wasmcloud.dev/component/net/wasihttp"
+)
 
 func init() {
-  httpserver := HttpServer{}
-  // Set the incoming handler struct to HttpServer
-  http.SetExportsWasiHttp0_2_0_IncomingHandler(httpserver)
-}
-
-func (h HttpServer) Handle(request HttpRequest, responseWriter HttpResponseWriter) {
-  // Construct HttpResponse to send back
-  headers := http.NewFields()
-  httpResponse := http.NewOutgoingResponse(headers)
-  httpResponse.SetStatusCode(200)
-
-  body := httpResponse.Body().Unwrap()
-  bodyWrite := body.Write().Unwrap()
-  bodyWrite.BlockingWriteAndFlush([]uint8("Hello from Go!\n")).Unwrap() // [!code --]
-  name := getNameFromPath(request.PathWithQuery().Unwrap()) // [!code ++]
-  bodyWrite.BlockingWriteAndFlush([]uint8(fmt.Sprintf("Hello %s!\n", name))).Unwrap() // [!code ++]
-
-  // Send HTTP response
-  okResponse := http.Ok[HttpOutgoingResponse, HttpError](httpResponse)
-  bodyWrite.Drop()
-  http.StaticOutgoingBodyFinish(body, http.None[http.WasiHttp0_2_0_TypesTrailers]())
-  http.StaticResponseOutparamSet(responseWriter, okResponse)
+	wasihttp.HandleFunc(handleRequest)
 }
 
-func getNameFromPath(path string) string { // [!code ++:8]
-  parts := strings.Split(path, "=")
-  if len(parts) == 2 {
-    return parts[1]
-  }
-  return "World"
+func handleRequest(w http.ResponseWriter, r *http.Request) {
+	name := r.URL.Query().Get("name") // [!code ++]
```

Comment:
`FormValue` ( which is not only for forms 🙈  )
https://pkg.go.dev/net/http#Request.FormValue

Reply:
Is this a more expected API? I'm good with the change, but even the comment about it not being just for forms makes me thing looking at the query string directly makes more sense. (I'm no complex Gopher, so, I'll take your word for whatever here)

Reply:
think is a personal preference.
I tend to use `FormValue` in examples to support both GET & POST and reduce surprises when copy/pasting into other codebases.
ex: going from GET handler to a POST handler with file upload ( content-type handling )

Reply:
I think the easy move to a POST request later on may be enough of an argument here 😄